NEO-TAT/tat_flutter
# Refactor Calendar page & Pump flutter version to 3.3.10

## Description 
Upgrade all dependencies and flutter sdk version to `3.3.10`.
The Android Kotlin version has upgraded to `1.8.0`.

There are three breaking changes:
- Awsome Dialog
  The dependency version has upgraded to v3, which deprecated the origin `DialogTyle` values.
  These update changes has addressed here: [3a0d35a](https://github.com/NEO-TAT/tat_flutter/pull/154/commits/3a0d35a6ab76d6dd1b4a526ae9ef97de773bd881).
- Alice
  Since the `alice` package on [`pub.dev`](https://pub.dev/packages/alice) depends on lots of its sub-packages which makes us hard to maintain TAT's dependency versions, we have migrated it to the [`alice_lightweight`](https://pub.dev/packages/alice_lightweight) package.
- Table Calendar
  Through the `table_calendar` package version has been upgraded to v3, we rebuilt the whole calendar page.
  First, a new `CalendarController`, which extends `GetXController`, has been created. We moved all non-UI layer functions and variables into this controller, and wrapped the calendar and calendar event list view widget by `GetBuilder` for receiving state updates from the controller.
  Then, we introduced a cache mechanism for the calendar data, which makes it not needed to re-fetch data while it has already exist.
  Additionally, the calendar style has also been refactored. You may found the event marker style changed, and the ripple effect of event ListTile has been fixed that it will not overflow of its container now.

In addition, the comments in the `pubspec.yml` file has been removed now, since they didn't express any useful meanings and make the file looks so messy.

> **Note**
> Includes some null safety migrations.

## How to Verify? 

- Calendar page
1. open calendar page, check if it shows today's event (if provided)
2. switch to different day in the same month, see if the event can be shown properly, and the calendar's selected day changes to that day.
3. swipe to next month (and previous month), see if it can fetches the data and show them (the marker) on the calendar right away.
4. repeat step 2 but now in step 3's month.
5. change the calendar format to `week`, `two week`
6. enable English UI and see if there are any strange things. 

- Debug page (Alice)
1. Do something that needs to fetch data from BE, such as `ischool page`.
2. go to dev page. (contact @Xanonymous-GitHub If you don't know where it is)
3. go to `Dio Log` page
4. click on each log, see if any error occurrs.

## Screenshots/GIF/Test Results (Optional) 
<img width="50%" alt="截圖 2022-12-30 上午6 30 31" src="https://user-images.githubusercontent.com/47718989/210018503-1ea2095a-07d2-4263-bc3a-54bc6298edc0.png"><img width="50%" alt="截圖 2022-12-30 上午6 30 13" src="https://user-images.githubusercontent.com/47718989/210018520-4eafa2c8-2627-47b5-884f-86a9c7af3ad3.png">
